### PR TITLE
Tickets/dm 30666

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: core/modules/replica/HttpModule.cc [hunk 1]
```diff
@@ -112,4 +113,33 @@ bool HttpModule::localLoadSecondaryIndex(string const& database) const {
...
+        auto const transactionInfo = databaseServices->transaction(transactionId);
+        database = transactionInfo.database;
+    }
+    debug(func, "database=" + database);
+
+    auto const databaseInfo = config->databaseInfo(database);
+    if (databaseInfo.isPublished) {
+        throw HttpError(
+                context() + "::" + func, "database '" + databaseInfo.name + " is already published.");
+    }
```
Comment (n8pease) [inline core/modules/replica/HttpModule.cc:141]: why is it an error to get database info for a published database?

Comment (iagaponenko) [reply core/modules/replica/HttpModule.cc:141]: The services that are using the method require that the database was still open for ingests (not _published_).

**Comment by n8pease** [comment]
The optional parameter to throw if published while getting database info is an improvement, but since it's defaulted to true it's still going to be non-obvious from where it's called that the method may throw for a not-get related reason. But I said I'd be ok with the option so I'll let it be.

**Comment by n8pease** [comment]
I thought there was a way to approve a PR but I can't find the button. Maybe because it's still failing travis? Anyway, I'm done, will mark the ticket.

**Pull request merged**